### ISSUE TITLE
🐛 Unclear radio button selection

### ISSUE THREAD

**Issue by BrookeCameron** [GCTC-NTGC/gc-digital-talent#13252]
## 🐛 Unclear radio button selection 

This issue was experienced when adding a government work experience to their profile.

### 🛡️ Fable testing results 

Difficult to use form controls (e.g. checkboxes)
I could not understand if the radio buttons were selected or not, when I clicked on it. I use high contrast mode with screen magnification. The radio button selection was not well communicated.

## 🦋 Expected Behaviour

form selections should be better communicated for users using high contrast mode 

## 🕵️ Details

- **Operating System**: Windows
- **Browser**: Chrome 
- **Assistive technology**: OS Screen magnification + high contrast 

## 📋 Steps to Reproduce

1. Turn on windows high contrast mode
2. Login as any user
3. Navigate to the personal information page
4. Edit the language profile
5. Select both checkboxes and radios
6. Observe you see no visual change inidcating selection

## 🙋‍♀️ Proposed Solution

Turn [off forced colour](https://developer.mozilla.org/en-US/docs/Web/CSS/forced-color-adjust) for specifically the before element.

## ✅ Acceptance Criteria

- [ ] Radio and checkboxes visually indicate selection in windows high contrast mode

**Comment by petertgiles** [comment]
I'm not sure what to do with this issue.  

The description starts with "form controls (e.g. checkboxes)" then describes radio buttons.  Is the scope all form controls, checkboxes, radio buttons, or both checkboxes and radio buttons? Why couldn't the tester understand if the radio button was selected?  Is is the colour?  The size?  They appear differently when using the AT?  Something with aria metadata?

Without understanding the problem, I don't know how to propose a solution.

**Comment by BrookeCameron** [comment]
_1.  Is the scope all form controls, checkboxes, radio buttons, or both checkboxes and radio buttons?_
Based on the additional comments I think there are issues with all the form controls. 

_2. Why couldn't the tester understand if the radio button was selected? Is is the colour? The size? They appear differently when using the AT? Something with aria metadata?_

### VoiceOver with Safari on iOS - Issue when selecting that the role was part of the government of Canada: 
When checking to make sure that I selected the correct button for choosing the role was part of the government of Canada, I am not able to tell that the button is selected without listening to the entire button description first. It would be much easier if, as soon as scrolling over the button, voiceover read selected before the description.

### TalkBack with Chrome on Android- Difficult to move focus between form fields:
When entering the start and end dates, the focus would move up to the start of the form once a selection had been made. This made it take longer to get through the form because I had to keep backtracking to fill out all the information.

**Comment by petertgiles** [comment]
> It would be much easier if, as soon as scrolling over the button, voiceover read selected before the description.

@esizer do you know if this is something that we can do?

> When entering the start and end dates, the focus would move up to the start of the form once a selection had been made. 

@esizer do you know what would cause this?  Is it parts of the form unhiding?  Can we mitigate it somehow?

**Comment by esizer** [comment]
> [@esizer](https://github.com/esizer) do you know if this is something that we can do?

Yeah, I have been trying to communicate this issue for a while now. At least, I rtihnk it is what I assumed but @BrookeCameron can correct me if I am wrong.

What happens with our radio buttons is it reads all of the text before it lets you know the state. We do not have control over the order it reads it in, that is the perogitve of the AT. However, what we do have contorl over and should stop doing, is writing full sentences (sometimes multiple) as the labels for these inputs 😅 

so, the fololwing input when selected would read:

> Checkbox, This was a role with the Government of Canada
Select this option if the employment was with a Government of Canada department, agency, crown corporation, or if you were a contractor working with one of these organizations. Checked.

![Image](https://github.com/user-attachments/assets/ff36514a-c765-4382-9cf4-54e1e0cb8031)

That is a lot of text to be read out before knowing what you selected 😢 

> [@esizer](https://github.com/esizer) do you know what would cause this? Is it parts of the form unhiding? Can we mitigate it somehow?

I would love to see a screen recording of this. My guess is something to do with conditional logic but can't be certain 🤔 We might need more info to reproduce.

**Comment by esizer** [comment]
> Yeah I'm on the same page as you [@esizer](https://github.com/esizer) this seems like the issue.

We have [a similar issue with our accordions as well](https://github.com/GCTC-NTGC/gc-digital-talent/issues/9156). Sadly, we have yet to address that issue either 😭

**Comment by BrookeCameron** [comment]
How can we move this @GCTC-NTGC/designers forward? @NienkeBr any thoughts on how we can fix this from a copy design perspective? Should we flag this as a reoccurring issue to Lauren?

**Comment by JoshBeveridge** [comment]
First, we should look into whether there are accessibility tools to properly label additional information. For example, we could test combining `aria-labelledby` in conjunction with `aria-describedby` to describe the form control and supplementary information respectively as proposed here: https://stackoverflow.com/questions/63694248/is-it-ok-to-combine-aria-label-and-aria-describedby

From my cursory research, we should test using `aria-details` based on its primary use case:
> However, when additional information, complex descriptions, or navigable content related to the object is necessary and available, the aria-details attribute should be used.
It also gives the user choice as to whether they want said details:
> When aria-details is included on an element, assistive technologies inform users of the availability of extended information, enabling the user to navigate to the referenced content.

Here's a link to its MDN page: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-details

**Comment by esizer** [comment]
> From my cursory research, we should test using `aria-details` based on its primary use case:

Unfortunately, safari does not support `aria-details` so it then becomes hidden entirely to voiceover users (which is where this issue was encountered). We are using `aria-describedby` since it has good support. However, as mentioned, it is combined with the label to be announced before the state.

**Comment by JoshBeveridge** [comment]
Safari is persistently a problem these days 😩

Removing the context isn't an option in most of these cases, because the text is required for the user to make an informed choice. I'm sure @NienkeBr can suggest shorter options, but they'll never be short enough to convey the same volume of information, which puts us back to square one.

At the end of the day, this is problematic yes, but it's not task blocking. Reducing the helpful information presented to all users in order to reduce the length of time a user has to listen to instruction seems like a bad compromise to me. All up for shortening the text where we can in instances where it's possible, but I don't think that's going to fix the problem wholesale.

**Comment by NienkeBr** [comment]
I agree with @JoshBeveridge and I would have liked it if there was a more technical issue that would work for all browsers (someone knows someone we can bug at safari?).
I'll have a go at shortening the options here.

**Comment by BrookeCameron** [comment]
Okay - it sounds like we won't be able to resolve the first concern. 

@esizer are you able to reproduce this (see below)  issue - if not would you like me to set up a test where we can get a screen recording with Fable? 

> TalkBack with Chrome on Android- Difficult to move focus between form fields:
> 
> When entering the start and end dates, the focus would move up to the start of the form once a selection had been made. This made it take longer to get through the form because I had to keep backtracking to fill out all the information.

**Comment by esizer** [comment]
> [@esizer](https://github.com/esizer) are you able to reproduce this (see below) issue - if not would you like me to set up a test where we can get a screen recording with Fable?

Yeah, sadly I was able to reproduce reliably. However, to complicate things further, this is not just the date fields or the experience form. I was able to get this to happen on most form fields across the site.

I tried it a few different times and it seems to be when collapsing the keyboard after entering text/numbers so hopefully that gets us closer?  🤔 

At first, I thought maybe this was a bug with talkback so I tried on different websites and sadly, those all worked fine so there seems to be something wrong on our site (maybe) 😭

**Comment by petertgiles** [comment]
@BrookeCameron it seems like the actionable scope of this issue has narrowed to just "focus jumps to the start of the form".  OK I close this issue and work with Eric to write up a new issue for the more narrow scope?

**Comment by esizer** [comment]
> [@BrookeCameron](https://github.com/BrookeCameron) it seems like the actionable scope of this issue has narrowed to just "focus jumps to the start of the form". OK I close this issue and work with Eric to write up a new issue for the more narrow scope?

I don't mean to speak for Brooke here, but I would not necessarily say that. We had a tangential conversation about that but the core issue here is that our radio group descriptions are being read out before the state. In an ideal world, the order would be:

 - Role, label, state, description

I have been looking into this and it may be related to our markcup here. I wouldn't hate to try changing it like so:

```tsx
// Current
<div>
  <label>
     <input />
     <span>{label}</span>
  </label>
  <div id="description">{description}</div>
</div>

// New
<div>
  <input />
  <label>{label}</label>
  <div id="description">{description}</div>
</div>
```

**Comment by petertgiles** [comment]
OK, so close this and write up two new issues?
